common/physics-system.cpp

```diff
@@ -367,18 +404,22 @@ namespace tec {
 	}
 
 	Position PhysicsSystem::GetPosition(eid entity_id) {
-		if (this->bodies.find(entity_id) != this->bodies.end() && this->bodies.at(entity_id)) {
-			auto pos = static_cast<CollisionBody*>(this->bodies.at(entity_id)->getUserPointer())->motion_state.transform.getOrigin();
+		auto body_iter = this->bodies.find(entity_id);
+		if (body_iter != this->bodies.end() && body_iter->second) {
+			//CollisionBody* body = static_cast<CollisionBody*>(body_iter->second->getUserPointer());
+			auto pos = body_iter->second->getWorldTransform().getOrigin();
```

Comment:
Changed this while debugging why bullet was giving me weird values, and then not giving me values. Turns out it's related to the simulation substeps setting, I left them this way because the game state pulls the transforms from here, and we want the non-interpolated versions, where as the custom motion state object gets the interpolated transform.
tl;dr: Feeding the MotionState transform back into Physics was breaking things